src/ProjectNeon/Assets/Features/Effects/Attack.cs

```diff
@@ -1,18 +1,42 @@
-﻿using System.Collections;
-using System.Collections.Generic;
+﻿
+
+
 using UnityEngine;
-using UnityEngine.Events;
 
-public sealed class Attack 
+public sealed class Attack  : Effect
 {
-    public Member Attacker { get; }
-    public Member Target { get; }
+    private Member _attacker;
+    public Member Attacker() { return this._attacker; }
+    private Target _target;
+    public Target Target() { return this._target; }
     public int Damage { get; }
 
-    public Attack(Member attacker, Member target, int damage)
+    public Attack(int damage)
     {
-        Attacker = attacker;
-        Target = target;
         Damage = damage;
     }
+
+    public void Apply(Member source, Target target)
+    {
+        _attacker = source;
+        _target = target;
+        if (target.Members.Length > 1)
+        {
+            target.Members.ForEach(
+                member => {
+                    new Attack(Damage).Apply(source, target);
+                }
+            );
+        } else
+        {
+            AllEffects.Create(
+                new EffectData { EffectType = EffectType.PhysicalDamage, FloatAmount = new FloatReference(Damage) }
+            ).Apply(source, target);
+            BattleEvent.Publish(this);
```

Comment:
Nothing should ever publish something of his own type.

Reply:
If you need this kind of behavior, separate it out into a System and a Message, so that one is pure data, and one is pure behavior.

Reply:
@SilasReinagel PLease see above

Reply:
This is a critical issue. We can't merge unless it's fixed. Nothing may publish himself. 

We shouldn't be publishing `Effects` at all. Only messages/events.

Reply:
@SilasReinagel DOne